src/classes/BusinessLogicServices/CarrierService.php

```diff
@@ -310,30 +311,58 @@ public function deletePacklinkCarriers()
     }
 
     /**
-     * Returns path to carrier logo or default carrier logo if logo for requested carrier doesn't exist.
+     * Gets the number of non-Packlink carriers.
      *
-     * @param string $carrierName Name of the carrier.
+     * @return int
+     */
+    public function getNumberOfOtherCarriers()
+    {
+        try {
```

Comment:
Why don't you return result here?

Reply:
It is returned at the end of the method. We try to have a single return point in methods.

Reply:
In this case I think we are loosing readability of code . We could improve code with one return but initiaitalizing result function to zero and change this value in try stament when needed . When we read the code firstly we need to calculate an array and at the end of code we calculate the result . We should use exceptions to capture errors and log

Reply:
updated